File: TarasDashboard/BusinessLogicLayer/Models/Response/SaleResponseModel.cs
using System.Collections.Generic;

namespace BusinessLogicLayer.Models.Response
{
    public class SaleResponseModel
    {
        public SaleOracleModel saleOracleModel { get; set; }
        public List<SaleStatisticModel> saleStatisticModels { get; set; }
        public List<SaleRegionsModel> saleRegionsModels { get; set; }
        public SaleRegionsModel lastLines { get; set; }
        public List<ExecutionPlanDate_HistoryModel> executionPlanDate_HistoryModels { get; set; }
        public List<DiagramModel> diagramModels { get; set; }
        public string Date {get; set;} 
        public string Time { get; set; }
        public bool Status { get; set; }
        public string Message { get; set; }

        public SaleResponseModel()
        {
            saleStatisticModels = new List<SaleStatisticModel>();
            saleRegionsModels = new List<SaleRegionsModel>();
            executionPlanDate_HistoryModels = new List<ExecutionPlanDate_HistoryModel>();
            diagramModels = new List<DiagramModel>();
        }
    }
}

File: TarasDashboard/TarasDashboard/ClientApp/src/app/models/response/sale.response.model.ts
import { SaleOracleModel } from "../sale.oracle.model";
import { SaleStatisticModel } from "../sale.statistic.model";
import { SaleRegionsModel } from "../sale.regions.model";
import { SaleHistoryModel } from "../sale.history.model";
import { DiagramModel } from "../diagram.model";
import { PlanSaleStockOnDateModel } from "../plan.sale.stock.on.date.model";
import { IndicatorsByNumberOfStoreModel } from "../indicators.by.number.of.store.model";

export interface SaleResponseModel{
    saleOracleModel?: SaleOracleModel;
    saleStatisticModels?: SaleStatisticModel[];
    saleRegionsModels?: SaleRegionsModel[];
    lastLines?: SaleRegionsModel;
    executionPlanDate_HistoryModels?: SaleHistoryModel[];
    diagramModels?: DiagramModel[];
    planSaleStockOnDateModels?: PlanSaleStockOnDateModel[];
    indicatorsByNumberOfStoreModel?: IndicatorsByNumberOfStoreModel;
    date?: string;
    time?: string;
    status?: boolean;
    message?: string;
}